interpreter/list_methods.c

```diff
@@ -430,3 +430,29 @@ int list_sort(struct YASL_State *S) {
 
 	return 0;
 }
+
+int list_insert(struct YASL_State *S) {
+	struct YASL_Object value = vm_pop((struct VM *) S);
+	yasl_int index = YASLX_checknint(S, "list.insert", 1);
+	struct YASL_List *ls = YASLX_checknlist(S, "list.insert", 0);
+	const yasl_int len = YASL_List_length(ls);
+
+	if (index > len) {
+		vm_print_err((struct VM *)S, "ValueError: %s index %d is out of bounds", "list.insert", index);
+		YASL_throw_err(S, YASL_VALUE_ERROR);
+	}
+	struct RC_UserData *new_list = rcls_new_sized(ls->size);
+	ud_setmt(new_list, S->vm.builtins_htable[Y_LIST]);
+	struct YASL_List *new_ls = (struct YASL_List *) new_list->data;
+	// Mutate the list mostly in place
+	ls->count += 1;
+	FOR_LIST(i, elmt, ls) {
+		if (i == (size_t) index) {
+			YASL_List_append(new_ls, value);
+		}
+		YASL_List_append(new_ls, elmt);
+	}
+	ls->items = new_ls->items;
```

Comment:
I suspect a dangling pointer can be left from the old list, or am I overthinking it

Reply:
Definitely not overthinking. Tests will catch it though, let me push some to your branch.